lib/lighthouse/benefits_claims/service.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'lighthouse/benefits_claims/configuration'
+require 'lighthouse/benefits_claims/service_exception'
+
+module BenefitsClaims
+  class Service < Common::Client::Base
+    configuration BenefitsClaims::Configuration
+    STATSD_KEY_PREFIX = 'api.benefits_claims'
+
+    # We don't want to show these in any of our UIs currently
+    # The majority of claims should not have any of these statuses
+    STATUS_DISALLOW_LIST = %w[CANCELED ERRORED PENDING].freeze
+
+    def initialize(icn)
+      @icn = icn
+      raise ArgumentError, 'no ICN passed in for LH API request.' if icn.blank?
+    end
+
+    def filter_claims(claim)
+      STATUS_DISALLOW_LIST.include?(claim.dig('attributes', 'status')) ? nil : claim
+    end
+
+    def get_claims
+      body = config.get("#{@icn}/claims").body
+
+      # Remove any claims with unwanted statuses
+      body['data'].map! { |claim| filter_claims(claim) }.compact
```

Comment:
I wrote this part based on this code: https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/claim_letters/claim_letter_downloader.rb#L60-L70. I think because of the `.map!` it doesn't matter if it's `.compact` or `.compact!`